app/lib/features/tasks/widgets/task_info.dart

```diff
@@ -360,25 +337,60 @@ class _TaskBodyState extends State<TaskBody> {
     return Stack(
       children: [
         Padding(
-          padding:
-              const EdgeInsets.only(left: 5, right: 5, bottom: 30, top: 10),
+          padding: const EdgeInsets.only(
+            left: 5,
+            right: 5,
+            bottom: 30,
+            top: 10,
+          ),
           child: child,
         ),
         Positioned(
           right: 0,
           bottom: 0,
           child: IconButton(
             key: TaskBody.editKey,
-            icon: const Icon(
-              Atlas.pencil_edit_thin,
-              size: 24,
-            ),
+            icon: const Icon(Atlas.pencil_edit_thin, size: 24),
             onPressed: () => setState(() => editMode = true),
           ),
         ),
       ],
     );
   }
+
+  Future<void> onSave() async {
+    final newBody = _textEditingController.text;
+    final description = widget.task.description();
+
+    final isNothing = description == null && newBody.isEmpty;
+    final isSame = description?.body() == newBody;
+    if (isNothing || isSame) {
+      // close and ignore, nothing actually changed
+      setState(() => editMode = false);
```

Comment:
this means it should return, no?
```suggestion
      setState(() => editMode = false);
      return;
```

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1580/commits/e2994e5ea5ff0de6244a032528607600fda43b00